Source/WebCore/platform/network/soup/NetworkStorageSessionSoup.cpp

```diff
@@ -410,6 +410,30 @@ void NetworkStorageSession::setCookie(const Cookie& cookie)
     soup_cookie_jar_add_cookie(cookieStorage(), cookie.toSoupCookie());
 }
 
+void NetworkStorageSession::setCookieJar(const Vector<Cookie>& cookies)
+{
+    SoupCookieJar* jar = cookieStorage();
+
+    // Delete existent cookies and add the new ones. During the process, disable
+    // signals from the cookie jar so we don't get a ton of them.
+    guint signalId = g_signal_lookup("changed", soup_cookie_jar_get_type());
+    GArray* blockedHandlerIdArray = g_array_new(FALSE, FALSE, sizeof(gulong));
+    gulong handlerId;
+    while ((handlerId = g_signal_handler_find(jar, (GSignalMatchType)(G_SIGNAL_MATCH_ID | G_SIGNAL_MATCH_UNBLOCKED), signalId, 0, NULL, NULL, NULL))) {
+        g_array_append_val(blockedHandlerIdArray, handlerId);
+        g_signal_handler_block(jar, handlerId);
```

Comment:
Even better, why not using `g_signal_handlers_block_matched()`, and then `g_signal_handlers_unblock_matched()`?

Reply:
`g_signal_handlers_block_matched` does not support G_SIGNAL_MATCH_ID (base on test and documentation: https://docs.gtk.org/gobject/func.signal_handlers_block_matched.html

But I found something similar which can be used and it works:
```
    guint signalId = g_signal_lookup("changed", soup_cookie_jar_get_type());
    gulong handler = g_signal_handler_find (jar, G_SIGNAL_MATCH_ID, signalId, 0, NULL, NULL, NULL);

    g_signal_handler_block(jar, handler);
```

I created also a patch to glib which adds a possibility to use  `g_signal_handlers_block_matched` with G_SIGNAL_MATCH_ID:
https://github.com/pgorszkowski-igalia/glib/commit/0eee1123840003f611cec484e5484ff6cf520e3d

I will try to contribute with this patch to glib project

Reply:
And one more thing which should be added here (after discussion with @aperezdc) : at the end of the `NetworkStorageSession::setCookieJar` one signal  "changed" should be sent.

Reply:
MR for glib change: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3376